### PR TITLE
Add Gherkin stories for group-level reports for an exercise/resource learners and Preview button

### PR DESCRIPTION
**Pull request by MisRob**
### Summary

Add Gherkin stories for
- Preview button for lesson resource/exercise reports
- group-level learner reports for lesson resource/exercise

### References

#5453 follow-up

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by radinamatic** [comment]
@MisRob Which branch should I use to test this? I tried both `develop` and `0.12.y` and can't seem to see the "View by groups" option...? :thinking:

**Comment by MisRob** [comment]
@radinamatic Hi Radina, it's available in `develop`, just checked it to be sure (also applies to exercises). It's on the resource report page in Lessons reports. Not sure what could be wrong, I'm thinking if I might have written that scenario in a misleading way maybe? I'm inserting screenshots so we're sure that you're on the page I implemented the feature for (Coach -> Reports -> Lessons tab -> lesson -> Report tab -> select resourcę)

Screenshot 2019-05-12 at 19 04 58

File: integration_testing/features/coach/coach-class-reports-lessons.feature [hunk 1]
```diff
@@ -72,6 +72,8 @@ Feature: Lessons subtab
       And I see the progress and average time spent columns for the resources
     When I click the <resource> resource
     Then I am on the *<resource>* report page
+      I see *Preview* button
```
Comment (jonboiser) [inline integration_testing/features/coach/coach-class-reports-lessons.feature]: Change to "And I see the *Preview* button"

File: integration_testing/features/coach/coach-class-reports-lessons.feature [hunk 2]
```diff
@@ -92,6 +94,33 @@ Feature: Lessons subtab
       # Clarify conditions for *Needs help*
       Then their *Progress* column states *Needs help*
 
+  Scenario: View the resource report page by groups
+    Given that I am on the <resource> report page
```
Comment (jonboiser) [inline integration_testing/features/coach/coach-class-reports-lessons.feature]: Should add an assumption that the classroom has groups.

I totally forgot to test this, but what happens if the classroom has no groups and you click the checkbox? I think both cases should have a Gherkin story

```
Given ...
    And the classroom has at least one learner group
    And the lesson has been assigned to either the entire class or at least one learner group


Given ...
   And the classroom has no learner groups
   And the lesson has been assigned to the entire class
```

Comment (MisRob) [reply integration_testing/features/coach/coach-class-reports-lessons.feature]: > I totally forgot to test this, but what happens if the classroom has no groups and you click the checkbox?

Basically, the same logic is used all the time => in this particular case when there are no groups at all, all learners will be displayed in "Ungrouped learners".

Screenshot 2019-05-14 at 18 45 45

Comment (MisRob) [reply integration_testing/features/coach/coach-class-reports-lessons.feature]: Agree that it makes sense to be more explicit on this in Gherkin stories, will update.

Comment (MisRob) [reply integration_testing/features/coach/coach-class-reports-lessons.feature]: Updated + I also extracted empty groups scenarios to their own stories to make things a bit more readable.

**Review by jonboiser — APPROVED** [review]
Looks good to me!

**Comment by MisRob** [comment]
Okay, thanks for your suggestions :) Will wait for @radinamatic to confirm if she was able to find it or no before merging.

**Comment by radinamatic** [comment]
False alarm @MisRob, I was looking at the wrong place :woman_facepalming: 

It seemed easier if I opened a PR on top of yours, hope you don't mind!

BTW, now that I look at coach feature files, there was already one [scenario that covered editing lesson details and managing resources _*from*_ Reports](https://github.com/learningequality/kolibri/blob/develop/integration_testing/features/coach/coach-edit-lesson-from-report.feature), so either delete those scenarios from this one, or delete the other file entirely, keeping both is superfluous and adds unnecessary tasks for the testers!

**Comment by MisRob** [comment]
@radinamatic No worries, thanks.

**Review by radinamatic — APPROVED** [review]
Yey, team work! :busts_in_silhouette: :+1:

**Pull request merged**